cdap-kafka/src/main/java/co/cask/cdap/kafka/run/KafkaServerMain.java

```diff
@@ -137,7 +160,10 @@ public void destroy() {
   }
 
   private Properties generateKafkaConfig(int brokerId, String zkConnectStr, String hostname, int port,
-                                         int numPartitions, int replicationFactor, String logDir) {
+                                         int numPartitions, int replicationFactor, String logDir, int retentionHours,
+                                         int bufferSendBytes, int bufferReceiveBytes, int socketRequestMaxBytes,
+                                         long intervalMessageSize, long messagePersistTime, int logFileSize,
+                                         int zookeeperTimeout) {
```

Comment:
Having multiple input parameters of the same type seems error prone. Can we pass cConf instead?